scripts/override_images/override_os_images.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Overrides a given list of OS images to a list with a single OS image
+that matches the latest release image in RELEASE_IMAGES
+"""
+import json
+import os
+import sys
+from argparse import ArgumentParser
+
+from assisted_service_client.models import OsImage, ReleaseImage
+
+from assisted_test_infra.test_infra import utils
+from consts import DEFAULT_CPU_ARCHITECTURE
+
+
+def get_os_image(
+    os_images: list[OsImage], openshift_version: str, cpu_architecture: str = DEFAULT_CPU_ARCHITECTURE
+) -> OsImage | None:
+    archs_images = [v for v in os_images if v.cpu_architecture == cpu_architecture]
+    os_images = [v for v in archs_images if v.openshift_version == openshift_version]
+    if len(os_images) >= 1:
+        return os_images[0]
+
+    return None
+
+
+def extract_version(release_image: str) -> str:
+    if not release_image:
+        return ""
+
+    full_ocp_version = utils.extract_version(release_image)
+    return f"{full_ocp_version.major}.{full_ocp_version.minor}"
+
+
+def main():
+    # Load default os images
+    with open(args.src, "r") as f:
+        os_images_parsed: list[OsImage] = json.load(f)
+
+    os_images = [OsImage(**os_image_dict) for os_image_dict in os_images_parsed]
+
+    if (release_images := os.getenv("RELEASE_IMAGES", "")) == "":
+        raise ValueError("RELEASE_IMAGES environment variable must be provided")
+
+    try:
+        release_images = json.loads(release_images)
+    except json.JSONDecodeError as e:
+        raise ValueError("RELEASE_IMAGES enironment variable content is invalid JSON") from e
+
+    if len(release_images) == 0:
+        raise ValueError("RELEASE_IMAGES enironment variable content must contain at least one release")
+
+    latest_release_image = ReleaseImage(**release_images[-1])
```

Comment:
This will return the last key not the latest release

Reply:
It works

Reply:
```
make run OPENSHIFT_VERSION=4.17
.
.
.
++ skipper run ./scripts/override_images/override_release_images.py --src ./assisted-service/data/default_release_images.json           
[skipper] Using build container: assisted-test-infra:latest                                                                             
+ RELEASE_IMAGES='[{"openshift_version":"4.17","cpu_architecture":"x86_64","cpu_architectures":["x86_64"],"url":"quay.io/openshift-release-dev/ocp-release:4.17.0-rc.5-x86_64","version":"4.17.0-rc.5","default":null,"support_level":"beta"}]'
+ export RELEASE_IMAGES
+ '[' kind == onprem ']'
+ '[' '' = '' ']'
+ '[' false '!=' true ']'
+ '[' '"false"' = true ']'
+ '[' '' '!=' '' ']'
+ '[' 4.17 '!=' '' ']'
++ skipper run ./scripts/override_images/override_os_images.py --src ./assisted-service/data/default_os_images.json
[skipper] Using build container: assisted-test-infra:latest
+ OS_IMAGES='[{"openshift_version":"4.17","cpu_architecture":"x86_64","url":"https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/pre-release/4.17.0-ec.3/rhcos-4.17.0-ec.3-x86_64-live.x86_64.iso","version":"417.94.202407010929-0"}]'
+ export OS_IMAGES

```

Reply:
I didn't say it doesn't -  you can't trust that  the last  key will be  the latest version

Reply:
I am not sure I follow. The least key is not the latest release, the order of actions is:

1. release_images[-1] returns some dictionary which represent a release image
2. ** unpacks the dictionary

Reply:
resolved